lighthouse-core/lib/i18n/i18n.js

```diff
@@ -130,10 +130,72 @@ function lookupLocale(locale) {
  * @param {Record<string, string | number>} [values]
  */
 function _preprocessMessageValues(icuMessage, values = {}) {
-  const clonedValues = JSON.parse(JSON.stringify(values));
+  let clonedValues = JSON.parse(JSON.stringify(values));
   const parsed = MessageParser.parse(icuMessage);
+
+  const elements = _collectAllCustomElementsFromICU(parsed.elements);
+
+  clonedValues = _processParsedElements(Array.from(elements.values()), clonedValues);
+
+  return clonedValues;
+}
+
+/**
+ * Function to retrieve all 'arguementElement's from an ICU message. An arguementElement
+ * is an ICU element that has an arguement attached like {varName, number, bytes}. This
+ * differs from 'messageElement's which are just arbitrary text in a message.
+ *
+ * Note: This function will recursively inspect plural elements for nested arguementElements.
+ *
+ * @param {Array<import('intl-messageformat-parser').Element>} parsedIcu
+ * @param {Map<string, import('intl-messageformat-parser').Element>} elements
+ */
+function _collectAllCustomElementsFromICU(parsedIcu, elements = new Map()) {
```

Comment:
it confused me a bit that both of these arguments are basically elements yet they're so differently named

- `currentElements` and `allElementsById`
- `icuElements` and `icuElementsById`
- `elements` and `previouslySeenElementsById`

idk any of those seem halfway decent?